lib/src/platform/src/platform.dart

```diff
@@ -18,27 +18,29 @@ class Platform {
   static final StreamsChannel streamsChannel =
       StreamsChannel('io.ably.flutter.stream', codec);
 
-  /// Initializing ably on platform side by invoking `register` platform method.
-  /// Register will clear any stale instances on platform.
-  static Future? _initializer;
+  /// This field will reset to true when the state is reset. This allows us to
+  /// if an app has been restarted, or a hot restart (not hot reload) has
+  /// happened.
+  static bool _shouldResetAblyClients = true;
```

Comment:
Could you use a non-static field here 🤔

Reply:
Not if everything else is static in this file 😂 . This conversation can be continued in the other: https://github.com/ably/ably-flutter/pull/219#discussion_r752149420

Reply:
I'm resolving this conversation on @ikbalkaya's behalf.